File: CPP/TranspositionCipher.cpp
#include <iostream>
#include <string>
using namespace std;

//O(n)
string transpositionCipher(string text) {
    string swapped = "";
    for (int i = 0; i < text.length() - 1; i += 2) {
        swapped += text[i + 1];
        swapped += text[i];
    }
    if (text.length() % 2 != 0) {
        swapped += text[text.length() - 1];
    }
    return swapped;
}

int main() {
    cout << transpositionCipher("COMPUTER") << endl;
    cout << transpositionCipher("PHONE") << endl;
}

File: Python/TranspositionCipher.py
#O(n)
def transposition_cipher(text: str) -> str:
    text_as_list = list(text)
    for i in range(0, len(text_as_list) - 1, 2):
        text_as_list[i], text_as_list[i + 1] = text_as_list[i + 1], text_as_list[i]
    return "".join(text_as_list)

print(transposition_cipher("COMPUTER"))
print(transposition_cipher("PHONE"))